_query-dsl/specialized/percolate.md

```diff
@@ -0,0 +1,620 @@
+        "_source": {
+          "query": {
+            "match": {
+              "title": "apple"
+            }
+          }
+        },
+        "fields": {
+          "_percolator_document_slot": [
+            0
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+## Percolate an existing indexed document
+
+Create a separate index for your documents:
+
+```json
+PUT /products
+{
+  "mappings": {
+    "properties": {
+      "title": {
+        "type": "text"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Add a document:
+
+```json
+POST /products/_doc/1
+{
+  "title": "Banana Smoothie Special"
+}
+```
+{% include copy-curl.html %}
+
+Check if the stored queries match the indexed document:
+
+```json
+POST /my_percolator_index/_search
+{
+  "query": {
+    "percolate": {
+      "field": "query",
+      "index": "products",
+      "id": "1"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The corresponding query is returned:
+
+```json
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.13076457,
+    "hits": [
+      {
+        "_index": "my_percolator_index",
+        "_id": "2",
+        "_score": 0.13076457,
+        "_source": {
+          "query": {
+            "match": {
+              "title": "banana"
+            }
+          }
+        },
+        "fields": {
+          "_percolator_document_slot": [
+            0
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+## Batch percolation (multiple documents)
+
+You can check multiple documents in one request:
+
+```json
+POST /my_percolator_index/_search
+{
+  "query": {
+    "percolate": {
+      "field": "query",
+      "documents": [
+        { "title": "Apple event coming soon" },
+        { "title": "Banana farms expand" },
+        { "title": "Cherry season starts" }
+      ]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Each match indicates which document slot matched using `_percolator_document_slot`:
+
+```json
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 0.46484798,
+    "hits": [
+      {
+        "_index": "my_percolator_index",
+        "_id": "2",
+        "_score": 0.46484798,
+        "_source": {
+          "query": {
+            "match": {
+              "title": "banana"
+            }
+          }
+        },
+        "fields": {
+          "_percolator_document_slot": [
+            1
+          ]
+        }
+      },
+      {
+        "_index": "my_percolator_index",
+        "_id": "1",
+        "_score": 0.41211313,
+        "_source": {
+          "query": {
+            "match": {
+              "title": "apple"
+            }
+          }
+        },
+        "fields": {
+          "_percolator_document_slot": [
+            0
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+## Multi-query percolation using named query
+
+You can percolate different documents inside a named query:
```

Comment:
We should also mention the comparative performance of this ways vs multiple documents in single percolate query

Reply:
Ah I see!, thanks